### ISSUE TITLE
A question about min output tokens

### ISSUE THREAD

**Issue by NaamaKadosh** [huggingface/text-generation-inference#764]
Hi,

The generate() method of AsyncClient gets a _max_new_tokens_ parameter to control the length of the output tokens. Why is not there a _min_new_tokens_ parameter? How can I request the generation of at least x tokens?

Thank you

**Comment by Narsil** [comment]
Hi @NaamaKadosh,

Not really.

Generation will just generate tokens indefinitely theoretically. `max_new_tokens` will bound that so that queries are not everlasting and don't consume infinite memory.

`min_new_tokens` doesn't really mean anything in that case.

However, generations can be cut short if the model decides to output `EOS` token. (It's a special token, most model have, which signal that the model is done outputting text). `text-generation-inference` simply follows that hint and stops.

The thing is that models are not really meant to output anything past EOS, so letting it continue will cause it to output garbage most likely.

`min_new_tokens` exist in the context of summarization but works by skewing the logits to forbid the model from outputting EOS, therefore forcing it to continue until EOS.

Instead of doing it the narrow way, we're thinking about how to enable many of such user defined manipulation of the model (guidance, token healing, templating). 

Is that clearer ?

**Comment by NaamaKadosh** [comment]
> Hi @NaamaKadosh,
> 
> Not really.
> 
> Generation will just generate tokens indefinitely theoretically. `max_new_tokens` will bound that so that queries are not everlasting and don't consume infinite memory.
> 
> `min_new_tokens` doesn't really mean anything in that case.
> 
> However, generations can be cut short if the model decides to output `EOS` token. (It's a special token, most model have, which signal that the model is done outputting text). `text-generation-inference` simply follows that hint and stops.
> 
> The thing is that models are not really meant to output anything past EOS, so letting it continue will cause it to output garbage most likely.
> 
> `min_new_tokens` exist in the context of summarization but works by skewing the logits to forbid the model from outputting EOS, therefore forcing it to continue until we allow it to output EOS.
> 
> Instead of doing it the narrow way, we're thinking about how to enable many of such user defined manipulation of the model (guidance, token healing, templating).
> 
> Is that clearer ?

It is,thank you!

**Comment by Narsil** [comment]
Ok, closing this then ! Cheers.
